### PR TITLE
Selinux role modules (for PMK, also viable for openstack if needed).

### PR DESCRIPTION
**Pull request by jayunit100**
Im adding more to this as we speak, but as is this should be mergeable without breaking anything, and provides a simple task that can be run  / integrated into express mainline at operator discretion.  Its based off the recipes living in https://github.com/platform9/selinuxmodules/blob/master/apply.sh , so please check that out for comprehensive docs.

### CODE REVIEW

**Comment by jayunit100** [comment]
This also changes the Vagrantfile to use Centos, since as of this PR, at least, the only thing that anyone using is Vagrant for is... centos and selinux :)

**Comment by jayunit100** [comment]
cc  @jb4free @mmccarre

**Comment by jayunit100** [comment]
cc @blake :)

File: roles/pmk-selinux/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+- copy:
+    src: "{{ role_path }}/files/pf9.te"
+    dest: /tmp/pf9.te
+- name: pmk-selinux
+  shell: |
```
Comment (blake) [inline roles/pmk-selinux/tasks/main.yml]: Try using the `yum` module to install packages. Also, avoid use of the `shell` module unless you need to use globbing, pipes, etc.

```yaml
- name: Install SELinux packages
  yum:
    state: present
    name: "{{ item }}"
  with_items:
    - checkpolicy
    - policycoreutils
    - policycoreutils-python

- name: Load SELinux policies
  command: "{{ item }}"
  args:
    chdir: /tmp
  with_items:
    - checkmodule -M -m -o pf9.mod pf9.te
    - semodule_package -o pf9.pp -m pf9.mod
    - semodule -i pf9.pp

File: roles/pmk-selinux/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+- copy:
+    src: "{{ role_path }}/files/pf9.te"
```
Comment (blake) [inline roles/pmk-selinux/tasks/main.yml]: You should not need `role_path`. Ansible will automatically look for pf9.te under files/ in the role's directory.

File: Vagrantfile_ubuntu [hunk 3]
```diff
@@ -0,0 +1,30 @@
+Vagrant.configure("2") do |config|
+  config.vm.box = "ubuntu/xenial64"
+
+  def create(config, hostname, ip)
+    config.vm.define hostname do |host|
+      host.vm.hostname = hostname
+      host.vm.network "private_network", ip: ip
+      host.vm.provision "shell", inline: "echo ubuntu:ubuntu | chpasswd"
+      host.vm.provision "shell", inline: "apt-get update && apt-get install -y python-minimal"
+      host.vm.provision "shell", inline: "yum update -y"
```
Comment (blake) [inline Vagrantfile_ubuntu]: Remove `yum` command.

File: Vagrantfile [hunk 4]
```diff
@@ -1,27 +1,35 @@
...
-  config.vm.box = "ubuntu/xenial64"
+  config.vm.box = "centos/7"
 
   def create(config, hostname, ip)
     config.vm.define hostname do |host|
       host.vm.hostname = hostname
       host.vm.network "private_network", ip: ip
-      host.vm.provision "shell", inline: "echo ubuntu:ubuntu | chpasswd"
-      host.vm.provision "shell", inline: "apt-get update && apt-get install -y python-minimal"
+ #     host.vm.provision "shell", inline: "echo ubuntu:ubuntu | chpasswd"
```
Comment (blake) [inline Vagrantfile]: Probably better to just remove these instead of commenting them out.

File: roles/pmk-selinux/tasks/main.yml (hunk 2 above)
Comment (jayunit100) [reply roles/pmk-selinux/tasks/main.yml]: interesting, i tried that - it wasn't copying them over automatically (vagrantfile has an example to repro).

File: roles/pmk-selinux/tasks/main.yml (hunk 1 above)
Comment (jayunit100) [reply roles/pmk-selinux/tasks/main.yml]: thanks ! done , pushing this up now

File: Vagrantfile_ubuntu (hunk 3 above)
Comment (jayunit100) [reply Vagrantfile_ubuntu]: good catchhh

File: Vagrantfile (hunk 4 above)
Comment (jayunit100) [reply Vagrantfile]: agree fixeddd

Comment (jayunit100) [reply Vagrantfile]: done thanks

**Pull request merged**